File: Swift/missing-number.swift
class Solution {
    func missingNumber1(_ nums: [Int]) -> Int {
        var set = Set<Int>()

        for num in nums {
            set.insert(num)
        }

        for i in 0...nums.count {
            if !set.contains(i) {
                return i
            }
        }

        fatalError("no valid outputs")
    }

    func missingNumber2(_ nums: [Int]) -> Int {
        return (0...nums.count).reduce(0, { return $0 + $1 }) - nums.reduce(0, { return $0 + $1 })
    }
}

File: Python/missing-number.py
class Solution:
    def missingNumber1(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        s = set()

        for num in nums:
            s.add(num)

        for i in range(len(nums) + 1):
            if not i in s:
                return i

        raise IOError("no valid outputs")

    def missingNumber2(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        return sum(range(len(nums) + 1)) - sum(nums)
